### ISSUE TITLE
Add IMAGE output (empty or transformed) not just preview

### ISSUE THREAD

**Issue by djdarcy** [djdarcy/ComfyUI-Smart-Resolution-Calc#4]
## Feature Request

Add a real IMAGE output (not just preview widget) that can output either an empty image (black/white/custom color) or the input image post-transform. This allows users to use the node for direct image manipulation without generating through KSampler.

## Current Behavior

**Preview Image Output:**
- Shows calculated dimensions visually in node
- Cannot be connected to other nodes
- Used only for visual feedback

**Latent Output:**
- Creates empty latent (black when decoded)
- Requires KSampler to generate actual content
- No way to output actual image data

## Proposed Behavior

**New IMAGE Output:**

Add optional IMAGE output with configurable content:

**Option 1: Empty Image**
- Generate empty image at calculated dimensions
- Configurable fill color (black/white/gray/custom RGB)
- Use cases: Test workflows, dimension verification, mask generation

**Option 2: Transform Input Image**
- When image input connected, output transformed version
- Apply scaling, cropping, padding based on settings
- Use cases: Batch resize, aspect ratio conversion, image preprocessing

**Option 3: Both (separate outputs)**
- `image_empty`: Empty image at calculated dimensions
- `image_transformed`: Input image transformed (only when connected)

### Empty Image Output
1. **Dimension testing**: Verify calculated dimensions before generation
2. **Mask generation**: Create empty masks at specific resolutions
3. **Workflow debugging**: Test connections without running expensive generation
4. **Batch processing**: Generate multiple empty images at different resolutions

### Transformed Image Output
1. **Batch resize**: Load image, resize to target, save
2. **Aspect ratio conversion**: Convert images between ratios with smart cropping
3. **Preprocessing pipeline**: Scale images before feeding to other nodes
4. **Direct manipulation**: Use dimension calculation for image transforms

### Implementation Options

**Option A: Single IMAGE output with mode selector**
```python
INPUT_TYPES = {
    "optional": {
        "image_output_mode": (["none", "empty", "transformed"],),
        "fill_color": ("STRING", {"default": "#000000"}),
    }
}

RETURN_TYPES = ("LATENT", "IMAGE", "INT", "INT", ...)
RETURN_NAMES = ("latent", "image", "width", "height", ...)
```

**Option B: Separate outputs**
```python
RETURN_TYPES = ("LATENT", "IMAGE", "IMAGE", "INT", "INT", ...)
RETURN_NAMES = ("latent", "image_empty", "image_transformed", "width", "height", ...)
```

**Recommendation**: **Option A** (mode selector) - Cleaner, less clutter, users choose what they need.

### Empty Image Generation

```python
import torch

def create_empty_image(width, height, color_hex="#000000"):
    # Parse hex color
    r = int(color_hex[1:3], 16) / 255.0
    g = int(color_hex[3:5], 16) / 255.0
    b = int(color_hex[5:7], 16) / 255.0
    
    # Create tensor [batch, height, width, channels]
    image = torch.zeros((1, height, width, 3))
    image[:, :, :, 0] = r
    image[:, :, :, 1] = g
    image[:, :, :, 2] = b
    
    return image
```

### Image Transform

When `image_output_mode = "transformed"` and image input connected:
```python
import torch.nn.functional as F

def transform_image(image, target_width, target_height, mode="scale"):
    # image shape: [batch, height, width, channels]
    b, h, w, c = image.shape
    
    if mode == "scale":
        # Simple resize
        # Permute to [batch, channels, height, width] for F.interpolate
        image = image.permute(0, 3, 1, 2)
        image = F.interpolate(image, size=(target_height, target_width), mode='bilinear')
        image = image.permute(0, 2, 3, 1)
    
    elif mode == "crop":
        # Crop to target aspect ratio, then scale
        # ... crop logic ...
    
    elif mode == "pad":
        # Pad to target aspect ratio, then scale
        # ... pad logic ...
    
    return image
```

### Widget UI

Add compact widget for image output configuration:
- Toggle: Enable IMAGE output (default OFF)
- Mode selector: none / empty / transformed
- Color picker: For empty mode (default black)

### Backward Compatibility

- IMAGE output optional (defaults to not connected)
- No changes to existing latent workflow
- Adding output doesn't break existing workflows

## Implementation Checklist

**Phase 1: Empty Image**
- [ ] Add `image_output_mode` parameter (default "none")
- [ ] Add `fill_color` parameter with color picker widget
- [ ] Implement `create_empty_image()` method
- [ ] Add IMAGE to RETURN_TYPES (optional output)
- [ ] Test with various dimensions and colors

**Phase 2: Transform Image**
- [ ] Implement "transformed" mode with scale/crop/pad
- [ ] Add transform mode selector widget
- [ ] Handle edge cases (no image connected, dimension mismatch)
- [ ] Test batch image transforms

**Phase 3: Polish**
- [ ] Add tooltips for new parameters
- [ ] Update documentation with IMAGE output examples
- [ ] Add workflow examples to README

## Questions

1. Should IMAGE output be always available or optional (toggle)?
2. Single output with mode selector or separate empty/transformed outputs?
3. What transform modes are most important (scale/crop/pad)?
4. Should color picker be hex string or RGB sliders?

## Priority

Medium - Extends node utility for image manipulation, separate from generation workflow.

## Related

- Issue #3 (Image to latent) - Shares transform logic
- Preview image widget (current implementation)
